### PR TITLE
[terra-paginator] Add meaningful ARIA context after page nagivation

### PR DESCRIPTION
**Pull request by chenyixuan625**
### Summary

**What was changed:**
Aria-live and aria-relevant could be used in a VisuallyHiddenText on the paginator container to contain a meaningful string that we construct. This visually hidden string will be read out by the screen reader when the user clicks any paginator button(so the paginator changes). This visually hidden string would be empty so nothing would be read out in the initial status.

**Why it was changed:**
Currently there are no ARIA contexts after the screen reader user navigates the pagination (change the page). This is not good from the Accessibility perspective and we would like to provide some meaningful context after the user click on the paginator buttons.

### Testing

https://github.com/cerner/terra-core/assets/44185348/806569ff-fc98-483f-ae58-ede6a94bb6c2

https://github.com/cerner/terra-core/assets/44185348/3acd37f2-713e-4978-a1cf-24c7c909e55f

This change was tested using:

- [ ] WDIO
- [x] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9504 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-core-docs/CHANGELOG.md [hunk 1]
```diff
@@ -7,7 +7,8 @@
 
 * Changed
   * Updated default search delay to 2500ms.
-  * Update Search field examples to be more functionality focused. 
+  * Update Search field examples to be more functionality focused.
```
Comment (zxeleven) [inline packages/terra-core-docs/CHANGELOG.md:10]:
```suggestion
  * Updated Search field examples to be more functionality focused.
```

File: packages/terra-paginator/src/ControlledPaginator.jsx [hunk 2]
```diff
@@ -206,11 +209,13 @@ class Paginator extends React.Component {
     const fullView = ariaLabelledBy
       ? (
         <nav className={cx('paginator', !totalCount && 'pageless', theme.className)} aria-labelledby={ariaLabelledBy}>
+          <VisuallyHiddenText aria-live="polite" aria-relevant="all" text={this.state.selectedPageMessage} />
```
Comment (zxeleven) [inline packages/terra-paginator/src/ControlledPaginator.jsx:212]: nit: can we move this to a const like `fullViewChildren`?

Same for the other paginators as well

File: packages/terra-paginator/translations/en.json [hunk 3]
```diff
@@ -4,6 +4,8 @@
   "Terra.paginator.first": "First",
   "Terra.paginator.last": "Last",
   "Terra.paginator.pageIndex": "Page {pageNumber}",
+  "Terra.paginator.pageSelected": "Page {pageNumber} selected",
```
Comment (vinaybhargavar) [inline packages/terra-paginator/translations/en.json:7]: nit: Do we have a Jira logged to get translations for other languages?

**Review by vinaybhargavar — APPROVED** [review]

File: packages/terra-core-docs/CHANGELOG.md (hunk 1 above)
Comment (chenyixuan625) [reply packages/terra-core-docs/CHANGELOG.md]: https://github.com/cerner/terra-core/pull/3886/commits/f0af8740cfd4e4ebb8736772976eb7dba772ad57#diff-0d87855fd9582c564e6ca6a9654fcfa4099b7e717a0704a990d80316f05bdf59R10

File: packages/terra-paginator/src/ControlledPaginator.jsx (hunk 2 above)
Comment (chenyixuan625) [reply packages/terra-paginator/src/ControlledPaginator.jsx]: https://github.com/cerner/terra-core/pull/3886/commits/f0af8740cfd4e4ebb8736772976eb7dba772ad57#diff-0bceee42b89777a164b9a100804131b9447d947169e02cffe25e8791c74a958aR209-R212

**Review by scottwilmarth — APPROVED** [review]

File: packages/terra-paginator/src/ControlledProgressivePaginator.jsx [hunk 4]
```diff
@@ -59,16 +59,32 @@ class ControlledProgressivePaginator extends React.Component {
...
       showReducedPaginator: false,
+      selectedPageMessage: '',
     };
   }
 
   handlePageChange(index) {
+    const {
+      intl,
+      totalCount,
+      itemCountPerPage,
```
Comment (sdadn) [inline packages/terra-paginator/src/ControlledProgressivePaginator.jsx:70]: nit: alphabetization

File: packages/terra-paginator/src/ControlledProgressivePaginator.jsx [hunk 5]
```diff
@@ -167,14 +183,20 @@ class ControlledProgressivePaginator extends React.Component {
       </>
     );
 
+    const navigationMessage = (
```
Comment (kenk2) [inline packages/terra-paginator/src/ControlledProgressivePaginator.jsx:186]: Same thing here from my previous comment.

File: packages/terra-paginator/tests/jest/ControlledPaginator.test.jsx
Comment (sdadn) [inline packages/terra-paginator/tests/jest/ControlledPaginator.test.jsx:1]: +1 on the test cases

File: packages/terra-paginator/tests/jest/ControlledPaginator.test.jsx [hunk 6]
```diff
@@ -34,6 +35,15 @@ describe('Paginator', () => {
     expect(wrapper).toMatchSnapshot();
   });
 
+  it('should render a Controlled Paginator with visually hidden text', () => {
+    const wrapper = shallowWithIntl(renderWithSelectedPage).dive();
+    wrapper.setState({ breakpoint: 'tiny' });
+    wrapper.update();
+    wrapper.find('PaginatorButton').first().simulate('click', { nativeEvent: { keyCode: null } });
+    expect(wrapper.find('VisuallyHiddenText').at(0).prop('text')).toEqual('Terra.paginator.pageSelected');
```
Comment (kenk2) [inline packages/terra-paginator/tests/jest/ControlledPaginator.test.jsx:43]: nit: Here and in other instances doesn't `.at(0)' mean the same thing as `.first()`?

Comment (chenyixuan625) [reply packages/terra-paginator/tests/jest/ControlledPaginator.test.jsx]: Used `.first()` to keep consistent: https://github.com/cerner/terra-core/pull/3886/commits/ec98741110d2d3a9ec49885806f8e14497ce5c0d#diff-e05da9d01a4643a4ec7a339a471781e030e36c6b9f8828137fe97345b6934c13R43

File: packages/terra-paginator/src/ControlledProgressivePaginator.jsx (hunk 4 above)
Comment (chenyixuan625) [reply packages/terra-paginator/src/ControlledProgressivePaginator.jsx]: Fixed: https://github.com/cerner/terra-core/pull/3886/commits/ec98741110d2d3a9ec49885806f8e14497ce5c0d#diff-a0973b9ee3044c6c2559965dcdbcc62bf0c707fb0c977ae1f59ca342ac87a7e3R74

File: packages/terra-paginator/src/ControlledPaginator.jsx [hunk 7]
```diff
@@ -289,14 +298,20 @@ class Paginator extends React.Component {
       </>
     );
 
+    const navigationMessage = (
```
[the remark this answers is not in the document]
Comment (chenyixuan625) [reply packages/terra-paginator/src/ControlledPaginator.jsx:298]: Refactored: https://github.com/cerner/terra-core/pull/3886/commits/ec98741110d2d3a9ec49885806f8e14497ce5c0d#diff-0bceee42b89777a164b9a100804131b9447d947169e02cffe25e8791c74a958aR213-R222
Please let me know if it looks good.

File: packages/terra-core-docs/src/terra-dev-site/doc/paginator/example/ControlledPaginatorExample.jsx [hunk 8]
```diff
@@ -40,8 +40,7 @@ class PaginatorExample extends React.Component {
         <Dialog
           header={(
             <h1 id="controlled_paginator_header_id">
-              Page
-              {this.state.currentPage}
```
[the remark this answers is not in the document]
Comment (chenyixuan625) [reply packages/terra-core-docs/src/terra-dev-site/doc/paginator/example/ControlledPaginatorExample.jsx:44]: This change is as per Michael's advice on the Accessibility office hour last week. The idea is that the paginator header is just labeling the contexts that likely won't change.

**Review by zxeleven — APPROVED** [review]

**Comment by sdadn** [comment]
Can you update  this with main and resolve conflicts?

File: packages/terra-paginator/translations/en.json (hunk 3 above)
Comment (chenyixuan625) [reply packages/terra-paginator/translations/en.json:7]: https://jira.cerner.com/browse/TRANS-14744

File: packages/terra-paginator/CHANGELOG.md [hunk 9]
```diff
@@ -11,6 +11,7 @@
 
 * Added
   * Added optional props `ariaLabelledBy` and `ariaLabel` to provide ARIA contexts for the paginator.
+  * Added a visually hidden text with meaningful contexts to help screen reader users understand when they navigate the pagination.
```
Comment (sdadn) [inline packages/terra-paginator/CHANGELOG.md:14]: This seems to be under the wrong section. Can you move it to `unreleased`?

File: packages/terra-core-docs/CHANGELOG.md [hunk 10]
```diff
@@ -6,6 +6,8 @@
 
 * Updated
   * Updated `terra-scroll` to add realistic examples.
+  *  Updated all paginator examples to use meaningful titles.
+  *  Updated testing recommendations to use `jest.spyOn` instead of `jest.mock`.
```
Comment (sdadn) [inline packages/terra-core-docs/CHANGELOG.md:10]: Updated with main & piggy-backing off this PR to moved this line to the correct section.

Comment (chenyixuan625) [reply packages/terra-core-docs/CHANGELOG.md:10]: Thank you for resolving the conflicts!

**Review by sycombs — APPROVED** [review]

**Review by sdadn — APPROVED** [review]

**Pull request merged**